rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +394,67 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED and \
+      participant_summary.samplesToIsolateDNA == SampleStatus.RECEIVED:
+
+      keys = ['sampleStatus%sTime' % test
+              for test in config.getSettingList(config.DNA_SAMPLE_TEST_CODES)]
+      sample_stored_time_list = \
+        [v for k, v in participant_summary if k in keys and v is not None]
+
+      sample_stored_time = min(sample_stored_time_list) if sample_stored_time_list else None
+
+      if sample_stored_time is not None:
+        return max([sample_stored_time,
+                   participant_summary.consentForElectronicHealthRecordsTime,
+                   participant_summary.questionnaireOnTheBasicsTime,
+                   participant_summary.questionnaireOnLifestyleTime,
+                   participant_summary.questionnaireOnOverallHealthTime,
+                   participant_summary.physicalMeasurementsFinalizedTime])
+      else:
+        return None
+    else:
+      return None
+
+  def calculate_core_ordered_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                         participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED:
+      keys = ['sampleOrderStatus%sTime' % test
+              for test in config.getSettingList(config.DNA_SAMPLE_TEST_CODES)]
+      sample_ordered_time_list = \
+        [v for k, v in participant_summary if k in keys and v is not None]
+
+      sample_ordered_time = min(sample_ordered_time_list) if sample_ordered_time_list else None
+
+      if sample_ordered_time is not None:
+        return max([sample_ordered_time,
```

Comment:
Similar to above, I'd add:

if participant_summary.enrollmentStatusCoreOrderedSampleTime:
return participant_summary.enrollmentStatusCoreOrderedSampleTime